### PR TITLE
Make file changes

### PR DESCRIPTION
**Pull request by prakharbhatt007**
# Mbed Edge pull request

`
Make files to build edge-core for building unit-test builds.
`

## Description

`This PR contains the makefiles needed to build the test build of edge-core`

## Test instructions

`Submitter: Please add instructions to test your PR here.`

`If your pull request is a fix for a defect please write instructions
to reproduce the defect.`

### API change(s)

 - [ ] Not applicable.
 - [ ] API is backwards compatible.
 - [ ] API documentation is updated.

### Example applications updated

 - [ ] Not applicable.

### CODE REVIEW

**Review by ygoyal18 — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	rm -rf build build-doc build-nondebug config/edge_version_info.h
+
+build:
+	mkdir -p build
+
+generate-cli-parsers:
+	cd edge-core && ./gen_docopt.sh
+
+lib/mbed-cloud-client/source/update_default_resources.c: lib/mbed-cloud-client
+	manifest-tool init -d "mbed.com" -m "mbed-edge" -q
```
Comment (OPpuolitaival) [inline Makefile:18]: This probably need some changes for manifest-tool 2.0 version.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	cd build && cmake -DFACTORY_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-doc:
+	mkdir -p build-doc && cd build-doc && cmake -DBUILD_DOCUMENTATION=ON .. && make edge-doc
+	echo "\033[0;33mDocumentation is at ./build-doc/doxygen/index.html\033[0m"
+
+run-edge-core:
+	./build/bin/edge-core -p 22225 -o 8080
+
+run-edge-core-valgrind:
```
Comment (OPpuolitaival) [inline Makefile:58]: Please remove this. Osmo tests are not public

**Review by OPpuolitaival — CHANGES_REQUESTED** [review]
Remove osmo test at least. I recommend also reading all commands and think do we really need those all?

File: Makefile (hunk 2 above)
Comment (prakharbhatt007) [reply Makefile]: I have removed that section.

File: Makefile (hunk 1 above)
Comment (ygoyal18) [reply Makefile:18]: Yes, this should be changed to `manifest-dev-tool`. Last time when I did that, the PR checks started to fail. I guess as we have not yet enabled the same tests in this repo's PR checks, its fine to make this change and fix what breaks later.

Comment (prakharbhatt007) [reply Makefile]: Done.

**Comment by ygoyal18** [comment]
@prakharbhatt007 mbed-cloud-client moved to git submodule change shouldn't be part of this PR.

**Comment by ygoyal18** [comment]
Closing this PR as Makefile will be added along with the unit test cases.

**Pull request closed without merging**